src/Command/MakeMigrationCommand.php

```diff
@@ -0,0 +1,105 @@
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+final class MakeMigrationCommand extends AbstractCommand
+{
+    protected static $defaultName = 'make:migrations';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Creates a migration with the diff or an empty migration if there are no diffs')
+            ->addOption('empty', null, InputOption::VALUE_OPTIONAL, 'Generate an empty classes', 'false')
+            ->addOption('db', null, InputOption::VALUE_REQUIRED, 'The database connection to use for this command.')
+            ->addOption('em', null, InputOption::VALUE_OPTIONAL, 'The entity manager to use for this command.')
+            ->addOption('shard', null, InputOption::VALUE_REQUIRED, 'The shard connection to use for this command.')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeMigrations.txt'))
+        ;
+    }
+
+    protected function writeNextStepsMessage(array $params, ConsoleStyle $io)
+    {
+        $io->text([
+            'Next: We just generated the migration',
+            'Find the documentation at <fg=yellow>http://symfony.com/doc/current/bundles/DoctrineMigrationsBundle/index.html</>'
+        ]);
+    }
+
+    protected function configureDependencies(DependencyBuilder $dependencies)
+    {
+        $dependencies->addClassDependency(
+            DoctrineCommand::class,
+            'migrations'
+        );
+    }
+
+    protected function getParameters(): array
+    {
+        return [];
+    }
+
+    protected function getFiles(array $params): array
+    {
+        return [];
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $params = $this->getParameters();
+
+        $isEmpty = $this->input->getOption('empty');
+        $options = ['db' => $this->input->getOption('db'), 'em' => $this->input->getOption('em'), 'shard' => $this->input->getOption('shard')];
+
+        try {
+            $diffMigrationCommand = $this->getApplication()->find('doctrine:migrations:diff');
+            $generateMigrationCommand = $this->getApplication()->find('doctrine:migrations:migrate');
+        } catch (CommandNotFoundException $e) {
+            throw new RuntimeCommandException(sprintf("Missing package%s: to use the migrations, run: \n\ncomposer require %s", 'migrations', 'migrations'));
+        }
+
+        if (true === $isEmpty) {
```

Comment:
is this needed?

```diff
- if (true === $isEmpty) {
+ if ($isEmpty) {
```

Reply:
I don't want it to be another type and equal to true.